File: Spring Boot/Courzelo/src/main/java/org/example/courzelo/dto/requests/module/ModuleRequest.java
package org.example.courzelo.dto.requests.module;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;
import java.util.Map;


@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class ModuleRequest {
    private String name;
    private String description;
    private String duration;
    private String semester;
    private Double scoreToPass;
    private Boolean isFinished;
    private List<String> skills;
    private int credit;
    private String program;
    private Map<String,Long> moduleParts;

}

File: Angular/src/app/shared/models/institution/ModuleRequest.ts
export interface ModuleRequest {
    name: string;
    description: string;
    skills: string[];
    semester: string;
    scoreToPass: number;
    duration: string;
    isFinished: boolean;
    credit: number;
    program: string;
    moduleParts: Map<string, number>;
}
